applications/nrf5340_audio/src/bluetooth/ble_audio_services.c

```diff
@@ -187,14 +191,19 @@ static void mcc_discover_mcs_cb(struct bt_conn *conn, int err)
 
 	LOG_DBG("mcc_discover_mcs_cb");
 
+	if (!mcs_discover_called) {
+		return;
+	}
+
```

Comment:
I believe the boolean should also be reset when the discovery has been completed (either successfully or with error).

Reply:
Agreed (had the same comment). IIRC the discover callback will also be called if disconnecting, as that would trigger a subscription error, so the call to `ble_mcs_discover_called_set` in the disconnected can probably be removed actually.

Reply:
I think the call in the disconnected can be kept - it is good for the cases where are are disconnected in the middle of a discovery.

Reply:
Changed

Reply:
Would still like your opinions on the case where we disconnect in the middle of discovery.  Still thinking that we should set the variable to false when disconnected, so that any auto-subscriptions at startup will be stopped in that case also.

Reply:
I have been informed that the callback will still be called in that case, in which case setting the variable on disconnect is not required.